src/hdmf/container.py

```diff
@@ -354,6 +354,13 @@ def parent(self, parent_container):
             if isinstance(parent_container, Container):
                 parent_container.__children.append(self)
                 parent_container.set_modified()
+            for child in self.children:
+                if type(child).__name__ == "DynamicTableRegion":
+                    if child.table.parent is None:
+                        msg = "The table for this DynamicTableRegion has not been added to the parent."
+                        warn(msg)
+                    else:
+                        continue
```

Comment:
Could you clarify why this is specific for `DynamicTableRegion`?  I'm just wondering whether we should more generally check for: 1) any scalar attribute that is specified with a reference dtype in the schema (which would include the above case of DynamicTableRegion) and 2) any child that is specified as a Link in the schema. In principle we could also check for 3) any dataset that has a reference dtype, but that may be too expensive in terms of the number of objects we would need to check.

Reply:
In the issue ticket #885 I voiced/pinged the question of addressing how to tackle the issue of links needing a warning when the container parent is set. The original issue comes from pynwb ElectrodesSeries having a link the the ElectrodesTable. This is just a DynamicTableRegion. I look more into hdmf and see that 
```
link_type = NamedTuple('DynamicTableLink',
--
1072 | [('source_table', DynamicTable),
1073 | ('source_column', Union[DynamicTableRegion, VectorIndex]),
1074 | ('target_table', DynamicTable)])

```
Seeing how the links are defined through DynamicTableRegion/VectorIndex and how VectorIndex is used in conjunction with DynamicTableRegion, I left the conditional to just check the former.

Anything beyond this I don't think falls within the original scope, but let's discuss further to see what else we can bring into the PR. @oruebel

Reply:
Thanks for the clarification. I'm fine with moving the discussion to a separate issue if that makes this PR too complex or would hold the PR back for too long from being merged. The PR as is, I think it useful as is and addresses a key use case, so I think it would be fine to merge and expand on it later. Up to you.

If you decide that we should leave the PR as is, then I think we should update the Changelog text to indicate that that this is for DynamicTableRegion. The current text refers to all links, which is really what led me to the question, because I was looking at the PR and not the original issue.

Reply:
I'll add the more general case to our discussions next week and let's merge this fix to tackle the issue.

Reply:
Sounds good.